File: backend/internal/model/sync_progress.go
package model

import "time"

// SyncProgress 同步进度表
type SyncProgress struct {
	ID            int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	SyncHistoryID int64     `gorm:"not null;index" json:"syncHistoryId"`
	Total         int       `gorm:"not null;default:0" json:"total"`
	Downloaded    int       `gorm:"not null;default:0" json:"downloaded"`
	Failed        int       `gorm:"not null;default:0" json:"failed"`
	CurrentFile   string    `gorm:"type:varchar(255)" json:"currentFile"`
	UpdatedAt     time.Time `gorm:"not null;default:CURRENT_TIMESTAMP" json:"updatedAt"`
}

func (SyncProgress) TableName() string { return "sync_progress" }

File: worker/src/db/models/sync_progress.rs
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "sync_progress")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,
    pub sync_history_id: i64,
    pub total: i32,
    pub downloaded: i32,
    pub failed: i32,
    #[sea_orm(column_type = "String(StringLen::N(255))", nullable)]
    pub current_file: Option<String>,
    pub updated_at: DateTimeWithTimeZone,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::sync_history::Entity",
        from = "Column::SyncHistoryId",
        to = "super::sync_history::Column::Id"
    )]
    SyncHistory,
}

impl Related<super::sync_history::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::SyncHistory.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
